src/server/auth/utils.js

```diff
@@ -31,6 +31,31 @@ export const getAccessibleResourcesFromArboristasync = async (jwt) => {
   return resources;
 };
 
+export const getAccessibleCreateResourcesFromArboristasync = async (jwt) => {
+  let data;
+  if (config.internalLocalTest) {
+    data = {
+      resources: [ // these are just for testing
+        '/programs/DEV/projects/test',
+        '/programs/jnkns/projects/jenkins',
+      ],
+    };
+  } else {
+    data = await arboristClient.listAuthorizedCreateResources(jwt);
+  }
+
+  log.debug('[authMiddleware] list create resources: ', JSON.stringify(data, null, 4));
+  if (data && data.error) {
+    // if user is not in arborist db, assume has no access to any
+    if (data.error.code === 404) {
+      return [];
+    }
+    throw new CodedError(data.error.code, data.error.message);
+  }
+  const resources = data.resources ? _.uniq(data.resources) : [];
+  return resources;
```

Comment:
We don't need to execute all those expensive array manipulations for this feature, if your goal is to determine whether am user is Guppy admin or not. This function can simply returns a boolean based on the return value from `listAuthorizedCreateResources()` 
Of couse variable and function names should be updated to relflect this change

Reply:
Ok boolean is returned when checking for refresh permissions